compiler/src/dmd/chkformat.d

```diff
@@ -827,13 +848,14 @@ enum Format
  *          even if `Format.error` is returned
  *      genSpecifier = Generic specifier. For instance, it will be set to `d` if the
  *           format is `hdd`.
+ *      scanfSpec = true if called from parseScanfFormatSpecifier
  *      useGNUExts = true if parsing GNU format extensions
  * Returns:
  *      Format
  */
 Format parseGenericFormatSpecifier(scope const char[] format,
-    ref size_t idx, out char genSpecifier, bool useGNUExts =
-    findCondition(global.versionids, Identifier.idPool("CRuntime_Glibc"))) nothrow pure @trusted
+    ref size_t idx, out char genSpecifier, bool scanfSpec = false,
+    bool useGNUExts = findCondition(global.versionids, Identifier.idPool("CRuntime_Glibc"))) nothrow pure @trusted
```

Comment:
The addition of the `scanfSpec` parameter means its strictly no longer "generic".

Reply:
Yeah, you can rename it to "common".

Reply:
I'm currently swaying towards having a `switch` in `parsePrintfFormatSpecifier` and `parseScanfFormatSpecifier` to handle any non-common formats, then fallback to this routine for the default case.

However, I got distracted by the fact that these functions are `@safe`, and somehow this code still compiles.  Then I found [this ancient bug](https://issues.dlang.org/show_bug.cgi?id=11048#c6).